src/offset.js

```diff
@@ -92,7 +92,12 @@ jQuery.fn.extend({
 
 		rect = elem.getBoundingClientRect();
 
-		// Make sure element is not hidden (display: none) or disconnected
+		// Return zeros for disconnected elements (gh-2310)
+		if ( !jQuery.contains( document, elem ) ) {
```

Comment:
We don't need or want to check ancestry, since the below width/height/getClientRects check already excludes these cases. Returning `rect` at the end of the function in place of an implicit `undefined` should be all that's required.